### DISCUSSION TITLE
[TOPAS_Brachy]:

### DISCUSSION THREAD

**Discussion by thanhtaiphys** [OpenTOPAS/OpenTOPAS#47] [Run-Problem] [answered]
### Contact Details

thanhtaiphys@gmail.com

### What was the issue?

Dear TOPAS Community,

I am learning how to **correctly** create an input for **TOPAS brachytherapy**. First, I have tried with the *"ViewAbdomen_rtdose"* example. 
After that, I modified this input for brachytherapy using patient data shared by Nguyen (who shared data in previous thread), but it is not working. Please see my input file attached [BrachyTopasCervix.txt](https://github.com/user-attachments/files/16146509/BrachyTopasCervix.txt). Could anyone look at my input file and correct it?

I have a few questions and would greatly appreciate your assistance:

# 1. Isocenter Positioning:

d:Ge/IsoCenterX   = 0. mm
d:Ge/IsoCenterY   = 0. mm
d:Ge/IsoCenterZ   = 0. mm

Where can I find these values to input into the above parameters?

# 2. Patient Positioning:

dc:Ge/Patient/DicomOriginX  = 0. mm
dc:Ge/Patient/DicomOriginY  = 0. mm
dc:Ge/Patient/DicomOriginZ  = 0. mm

The same question as above, how do I get these values?

# 3. Source Position:
I am trying to ensure that the source in my simulation matches the exact position as defined in the TPS plan. Despite various adjustments, the source does not align correctly with the patient's position as defined in the DICOM files. Could you provide guidance on how to accurately position the source in the simulation according to the TPS plan?

# 4. Dose Scaling Factor:
If the simulation is correct, I will get **RTdose in DICOM format**. How can I get the **absolute dose from the Monte Carlo** simulation to compare with the TPS? Do we need to calculate a dose scaling factor that converts the MC dose to the absolute dose?

Thank you in advance for your help!
Duong

### TOPAS category

Geometry

### Version

4.0

### What Operative System (OS) are you running on?

Windows (WSL)

### Relevant log output

```shell
tduong@TINGLEE14:~/Applications/thanhtai/Brachytherapy$ topas BrachyTopasCervix.txt

Welcome to TOPAS, Tool for Particle Simulation (Version 4.0)
Loading parameters starting from: BrachyTopasCervix.txt

Geant4 Data directory has been specified by the
TOPAS_G4_DATA_DIR environment variable as /home/tduong/Applications/GEANT4/G4DATA

**************************************************************
 Geant4 version Name: geant4-11-01-patch-03 [MT]   (10-November-2023)
  << in Multi-threaded mode >> 
                       Copyright : Geant4 Collaboration
                      References : NIM A 506 (2003), 250-303
                                 : IEEE-TNS 53 (2006), 270-278
                                 : NIM A 835 (2016), 186-225
                             WWW : http://geant4.org/
**************************************************************

TOPAS is in MT mode, setting number of threads to: 24

TOPAS set the tolerances based on the World size to:
      Surface tolerance = 0.006 nm, and radial tolerance = 0.006 nm.
Registered use of changeable parameter: Tf/PositionX/Value  by component: Capsule
Registered use of changeable parameter: Tf/PositionY/Value  by component: Capsule
Registered use of changeable parameter: Tf/PositionZ/Value  by component: Capsule
Registered use of changeable parameter: Ge/Patient/DicomOriginX  by component: Patient
Registered use of changeable parameter: Ge/Patient/DicomOriginY  by component: Patient
Registered use of changeable parameter: Ge/Patient/DicomOriginZ  by component: Patient
TOPAS will read DICOM directory: Case2_NTTH_Fraction1
RTDOSE file is found: Case2_NTTH_Fraction1/RD.1.3.6.1.4.1.2452.6.607654830.1313807156.2891959176.2218885571.dcm
  # of Voxels: (63, 64, 96) 
  Voxel Dimensions X,Y,Z [mm]: (1, 1, 1)
  ImagePositionPatient (mm): (-33.718,-169.101,-1080.27)
  DoseGrid center (mm):(-2.718,-137.601,-1032.77)
  Image size (bytes): 774144
  DoseUnit:GY
  DoseScaling:0.000305204
    Iso Center [cm]:       ( 0.7, -16, -99.7 )
    # of Voxels:           ( 512, 512, 84 )
    Voxel Dimensions X and Y [cm]: ( 0.0699219, 0.0699219 ) 
    Number of Slice Thickness Sections: 1
      Slice thickness section 1 has number of voxels: 84 of thickness: 0.3 cm.
    Rotation [deg]:       ( 0, 0, 0 )
Checking overlaps for volume Patient:0 (G4Box) ... OK! 

Creating material map from image: Case2_NTTH_Fraction1
Found Imaging Value: -1024 less than minimum value in conversion table: -1000
Resetting value to minimum value from the table: -1000
This message is only written once per patient file. There may also be other underflow values. 
Found Imaging Value: 3065 greater than maximum value in conversion table: 2995
Resetting value to maximum value from the table: 2995
This message is only written once per patient file. There may also be other overflow values. 
Finished defining materials from imaging. Total number of materials used was: 2990
Creating parallel world: Patient/RTDoseGrid
Checking overlaps for volume Patient/RTDoseGrid:0 (G4Box) ... OK! 
Creating parallel world: Capsule
Checking overlaps for volume CableStainlessSteelCylinder:0 (G4Tubs) ... OK! 
Checking overlaps for volume CenterStainlessSteelCylinder:0 (G4Tubs) ... OK! 
Checking overlaps for volume AirFilling:0 (G4Tubs) ... OK! 
Checking overlaps for volume IridiumCore:0 (G4Tubs) ... OK! 
Checking overlaps for volume FullStainlessSteelCylinder:0 (G4Tubs) ... OK! 
Checking overlaps for volume LeftEndCone:0 (G4Cons) ... OK! 
Checking overlaps for volume RightStainlessSteelCone:0 (G4Cons) ... OK! 

Checking for Unused Components.
These are Components that were never built even though they were at least
partially defined in some parameters. This is not necessarily an error.
It may be an intentional choice to have these Component definitions handy for later use.
To turn off this check, set Ge/CheckForUnusedComponents to False
  The Parent of this Component was never specified: ActiveSource
  The Parent of this Component was never specified: StainlessSteel

Registered use of changeable parameter: So/ActiveSource/NumberOfHistoriesInRun  by source: ActiveSource
Visualization verbosity changed to quiet (0)

TimeFeature updated: tf/histories/value = 10 at: 0 (ms)

TimeFeature updated: tf/historiesweight/value = 305 at: 0 (ms)

TimeFeature updated: tf/positionx/value = -3.7 mm at: 0 (ms)

TimeFeature updated: tf/positiony/value = -1015.4 mm at: 0 (ms)

TimeFeature updated: tf/positionz/value = 139.1 mm at: 0 (ms)

-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : GeomMgt0002
      issued by : G4SmartVoxelHeader::BuildNodes()
PANIC! - Overlapping daughter with mother volume.
         Daughter physical volume Patient
         is entirely outside mother logical volume World !!
*** Fatal Exception *** core dump ***
 **** Track information is not available at this moment
 **** Step information is not available at this moment

-------- EEEE -------- G4Exception-END --------- EEEE -------


*** G4Exception: Aborting execution ***
/home/tduong/shellScripts/topas: line 4: 14961 Aborted                 (core dumped) ~/Applications/TOPAS/OpenTOPAS-install/bin/topas $1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
Hi there, regarding your questions:
1. Isocenter Position. This is embedded in your DICOM-CT, it is also displayed at the beginning of the simulation, see also this response https://github.com/OpenTOPAS/OpenTOPAS/discussions/28#discussioncomment-9874026
2. Patient Positioning. This is automatically computed by Topas (see the link above)
3. Visualization is quite helpful here! But it takes a while to render and confirm.
4. The scorers DoseToMedium and DoseToMaterial have a new unitless parameter `u:Sc/MyScorer/OutputWeightingFactor`, so that you can scale the entire dose distribution. But you must first find this value by your own. E.g. get the dose per primary history at reference conditions of your brachytherapy system (read AAPM-TG43) and associate it with your prescribed dose. Then, use this as scaling factor. 
Good luck!

**Reply by thanhtaiphys** [reply]
Dear Dr. Mendez,

Thank you so much for your quick and helpful response! I will review your suggestions and reach out again if I have further questions.

I also want to express my gratitude for creating this forum for efficient communication. I really appreciate your assistance.

All the best,
Duong
